src/internal/_dispatchToMapMethod.js

```diff
@@ -0,0 +1,10 @@
+var _mapFunctions = require('./_mapFunctions');
+var _objectFunctions = require('./_objectFunctions');
+
+
+module.exports = function _dispatchToMapMethod(methodname) {
+  return function() {
+    var isMap = typeof Map === 'function' && arguments[arguments.length - 1] instanceof Map;
+    return (isMap ? _mapFunctions : _objectFunctions)[methodname].apply(null, arguments);
+  };
+};
```

Comment:
Or possibly the untested

```js
module.exports = (function() {
    return typeof Map === 'function' ? function _dispatchToMapMethod(methodname) {
        return function () {
            var isMap = arguments[arguments.length - 1] instanceof Map;
            return (isMap ? _mapFunctions : _objectFunctions)[methodname].apply(null, arguments);
        };
    } : function _callObjectMethod(methodname) {
        return _objectFunctions[methodname].apply(null, arguments);
    }
}());
```

As I'm not sure we want to check `typeof Map === 'function'` on every call.